src/posix/platform/infra_if.cpp

```diff
@@ -0,0 +1,355 @@
+/*
+ *  Copyright (c) 2020, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file implements the infrastructure interface for posix.
+ */
+
+#include "platform-posix.h"
+
+#if OPENTHREAD_CONFIG_BORDER_ROUTING_ENABLE
+
+#ifdef __APPLE__
+#define __APPLE_USE_RFC_3542
+#endif
+
+#include <errno.h>
+#include <ifaddrs.h>
+// clang-format off
+#include <netinet/in.h>
+#include <netinet/icmp6.h>
+// clang-format on
+#include <sys/types.h>
+#include <unistd.h>
+
+#include <openthread/platform/infra_if.h>
+
+#include "common/code_utils.hpp"
+
+static char         sInfraIfName[IFNAMSIZ] = "UNKNOWN";
+static uint32_t     sInfraIfIndex          = 0;
+static int          sInfraIfIcmp6Socket    = -1;
+static otIp6Address sInfraIfLinkLocalAddr;
+
+otError otPlatInfraIfSendIcmp6(uint32_t            aInfraIfIndex,
```

Comment:
```suggestion
otError otPlatInfraIfSendIcmp6(unsigned int            aInfraIfIndex,
```

Reply:
Yes, linux [<net/if.h>](https://linux.die.net/man/3/if_indextoname) is defining the index as `unsigned int`. But to notice that this is a public OT platform API and I think the API itself should be "platform-agnostic". For posix, the interface index is `unsigned int`, for other platforms, it may be `unsigned short` ( or `int` and `short` have the same length). The API should support the maximum length and I think `uintxx_t` types are for eliminating such issues (At lease, other OT platform APIs are using fixed-length types). How do you think?

Reply:
I think we used uint32_t for interface indexes in other places of code.